diff -ruN target/linux/ramips/base-files/etc/board.d/02_network target/linux/ramips/base-files/etc/board.d/02_network
--- target/linux/ramips/base-files/etc/board.d/02_network	2021-01-25 09:29:08.901370257 +0500
+++ target/linux/ramips/base-files/etc/board.d/02_network	2021-01-18 11:25:53.756674000 +0500
@@ -173,6 +173,10 @@
 	netgear,r6120|\
 	r6220|\
 	netgear,r6350|\
+	beeline,sbtplus)
+		ucidef_add_switch "switch0" \
+			"1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4" "0:wan" "6@eth0"
+		;;	
 	wndr3700v5)
 		ucidef_add_switch "switch0" \
 			"0:lan:4" "1:lan:3" "2:lan:2" "3:lan:1" "4:wan" "6@eth0"
diff -ruN target/linux/ramips/base-files/lib/upgrade/platform.sh target/linux/ramips/base-files/lib/upgrade/platform.sh
--- target/linux/ramips/base-files/lib/upgrade/platform.sh	2021-01-25 09:29:08.901370257 +0500
+++ target/linux/ramips/base-files/lib/upgrade/platform.sh	2020-11-23 10:52:17.519254000 +0500
@@ -58,6 +58,7 @@
 	hc5962|\
 	r6220|\
 	netgear,r6350|\
+	beeline,sbtplus|\
 	ubnt-erx|\
 	ubnt-erx-sfp|\
 	xiaomi,mir3g|\
diff -ruN target/linux/ramips/dts/GHL-R-001.dts target/linux/ramips/dts/GHL-R-001.dts
--- target/linux/ramips/dts/GHL-R-001.dts	2021-01-25 09:29:08.905370342 +0500
+++ target/linux/ramips/dts/GHL-R-001.dts	2021-01-18 16:55:17.208809000 +0500
@@ -47,7 +47,7 @@
 	};
 };
 
-&uartlite3 {
+&uart3 {
 	status = "okay";
 };
 
diff -ruN target/linux/ramips/dts/mt7621.dtsi target/linux/ramips/dts/mt7621.dtsi
--- target/linux/ramips/dts/mt7621.dtsi	2021-01-25 09:29:08.917370598 +0500
+++ target/linux/ramips/dts/mt7621.dtsi	2021-01-24 14:58:10.000000000 +0500
@@ -32,6 +32,8 @@
 
 	aliases {
 		serial0 = &uartlite;
+		serial2 = &uartlite2;
+		serial3 = &uartlite3;
 	};
 
 	pll: pll {
@@ -170,51 +172,41 @@
 		uartlite: uartlite@c00 {
 			compatible = "ns16550a";
 			reg = <0xc00 0x100>;
-
 			clock-frequency = <50000000>;
-
 			interrupt-parent = <&gic>;
 			interrupts = <GIC_SHARED 26 IRQ_TYPE_LEVEL_HIGH>;
-
 			reg-shift = <2>;
 			reg-io-width = <4>;
 			no-loopback-test;
 		};
 
-		uartlite2: uartlite2@d00 {
+		uartlite2: uart2@d00 {
 			compatible = "ns16550a";
 			reg = <0xd00 0x100>;
-
 			clock-frequency = <50000000>;
-
 			interrupt-parent = <&gic>;
 			interrupts = <GIC_SHARED 27 IRQ_TYPE_LEVEL_HIGH>;
-
 			reg-shift = <2>;
 			reg-io-width = <4>;
-
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart2_pins>;
-
 			status = "disabled";
 		};
 
-		uartlite3: uartlite3@e00 {
+		uartlite3: uart3@e00 {
 			compatible = "ns16550a";
 			reg = <0xe00 0x100>;
-
+			clocks = <&sysclock>;
 			clock-frequency = <50000000>;
-
 			interrupt-parent = <&gic>;
 			interrupts = <GIC_SHARED 28 IRQ_TYPE_LEVEL_HIGH>;
-
 			reg-shift = <2>;
 			reg-io-width = <4>;
-
+			no-loopback-test;
+			current-speed = <115200>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart3_pins>;
-
-			status = "disabled";
+			status = "okay";
 		};
 
 		spi0: spi@b00 {
@@ -266,7 +258,7 @@
 			#dma-channels = <1>;
 			#dma-requests = <1>;
 
-			status = "disabled";
+			status = "okay";
 		};
 	};
 
diff -ruN target/linux/ramips/dts/QUAD-E4G.dts target/linux/ramips/dts/QUAD-E4G.dts
--- target/linux/ramips/dts/QUAD-E4G.dts	2021-01-25 09:29:08.909370428 +0500
+++ target/linux/ramips/dts/QUAD-E4G.dts	2021-01-18 16:54:46.564219000 +0500
@@ -227,10 +227,10 @@
 	};
 };
 
-&uartlite2 {
+&uart2 {
 	status = "okay";
 };
 
-&uartlite3 {
+&uart3 {
 	status = "okay";
 };
diff -ruN target/linux/ramips/dts/SBTplus.dts target/linux/ramips/dts/SBTplus.dts
--- target/linux/ramips/dts/SBTplus.dts	1970-01-01 05:00:00.000000000 +0500
+++ target/linux/ramips/dts/SBTplus.dts	2021-01-24 16:08:44.000000000 +0500
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "beeline,sbtplus", "mediatek,mt7621-soc";
+	model = "SmartBox Turbo+";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "sbtplus:green:power";
+			gpios = <&gpio0 15 GPIO_ACTIVE_HIGH>;
+		};
+
+
+		usb {
+			label = "sbtplus:blue:usb";
+			gpios = <&gpio0 13 GPIO_ACTIVE_HIGH>;
+			trigger-sources = <&xhci_ehci_port1>, <&ehci_port2>;
+			linux,default-trigger = "usbport";
+		};
+
+		wifi {
+			label = "sbtplus:red:wifi";
+			gpios = <&gpio0 16 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+		wps {
+			label = "wps";
+			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		usbpower {
+			gpio-export,name = "usbpower";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x100000>;
+			read-only;
+		};
+
+		partition@100000 {
+			label = "SC PART_MAP";
+			reg = <0x100000 0x100000>;
+			read-only;
+		};
+
+		partition@800000 {
+			label = "kernel";
+			reg = <0x200000 0x400000>;
+		};
+
+		partition@600000 {
+			label = "ubi";
+			reg = <0x600000 0x7800000>;
+		};
+
+		factory: partition@7e00000 {
+			label = "factory";
+			reg = <0x7e00000 0x100000>;
+			read-only;
+		};
+
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		reg = <0x0 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x4>;
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "uart2", "jtag", "wdt";
+			ralink,function = "gpio";
+		};
+	};
+};
diff -ruN target/linux/ramips/image/mt7621.mk target/linux/ramips/image/mt7621.mk
--- target/linux/ramips/image/mt7621.mk	2021-01-25 09:29:08.929370853 +0500
+++ target/linux/ramips/image/mt7621.mk	2021-01-18 11:23:39.326102000 +0500
@@ -389,6 +389,28 @@
 endef
 TARGET_DEVICES += netgear_r6350
 
+define Device/beeline_sbtplus
+  DTS := SBTplus
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4096k
+  IMAGE_SIZE := 40960k
+  UBINIZE_OPTS := -E 5
+  SERCOMM_HWID := CHJ
+  SERCOMM_HWVER := A001
+  SERCOMM_SWVER := 0x0052
+  IMAGES += factory.img kernel.bin rootfs.bin
+  IMAGE/factory.img := pad-extra 1024k | append-kernel | pad-to 6144k | append-ubi | \
+	pad-to $$$$(BLOCKSIZE) | sercom-footer | pad-to 128 | zip $$$$(DEVICE_MODEL).bin | sercom-seal
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-ubi | check-size $$$$(IMAGE_SIZE)
+  DEVICE_TITLE := Beeline SmartBox Turbo+
+  DEVICE_PACKAGES := \
+	kmod-mt7603 kmod-mt7615e kmod-usb3 kmod-usb-ledtrig-usbport wpad-basic
+endef
+TARGET_DEVICES += beeline_sbtplus
+
 define Device/MikroTik
   BLOCKSIZE := 64k
   IMAGE_SIZE := 16128k
